packages/@react-stately/datepicker/src/useDateFieldState.ts

```diff
@@ -156,16 +157,27 @@ export function useDateFieldState<T extends DateValue = DateValue>(props: DateFi
     throw new Error('Invalid granularity ' + granularity + ' for value ' + v.toString());
   }
 
-  let defaultFormatter = useMemo(() => new DateFormatter(locale), [locale]);
-  let calendar = useMemo(() => createCalendar(defaultFormatter.resolvedOptions().calendar), [createCalendar, defaultFormatter]);
-
-  let [value, setDate] = useControlledState<DateValue>(
+  let [value, setDate2] = useControlledState<DateValue>(
     props.value,
     props.defaultValue,
     props.onChange
   );
 
-  let calendarValue = useMemo(() => convertValue(value, calendar), [value, calendar]);
+  let setDate = (v) => {
+    console.log('wrappingSetDate: clone, v', Object.assign({}, v), v);
+    setDate2(v);
+  }
+
+  let defaultFormatter = useMemo(() => new DateFormatter(locale), [locale]);
+  let calendar = useMemo(() => {
+    console.log('useMemo: defaultFormatter.resolvedOptions(), value', defaultFormatter.resolvedOptions());
+    return createCalendar(defaultFormatter.resolvedOptions().calendar);
+  }, [createCalendar, defaultFormatter]);
+
+  let calendarValue = useMemo(() => {
+    console.log('useMemo: value, calendar, convert', value, calendar, convertValue(value, calendar));
```

Comment:
After `value` is updated I wanted to see how it was mutated, but found this was getting an unexpected value.

Also this and the useMemo above get called a few times each after a value update and I'd only expect once.